src/test/java/io/camunda/zeebe/bpmnassert/util/Utilities.java

```diff
@@ -146,22 +149,26 @@ public static PublishMessageResponse sendMessage(
     return response;
   }
 
-  public static void increaseTime(final ZeebeEngine engine, final Duration duration) {
-    engine.clock().increaseTime(duration);
+  public static void increaseTime(final InMemoryEngine engine, final Duration duration) {
+    engine.increaseTime(duration);
     waitForIdleState(engine);
   }
 
   public static void completeTask(
-      final ZeebeEngine engine, final ZeebeClient client, final String elementId) {
-    Record<JobRecordValue> lastRecord = null;
-    for (final Record<JobRecordValue> record : engine.jobRecords().withElementId(elementId)) {
-      if (record.getIntent().equals(JobIntent.CREATED)) {
-        lastRecord = record;
-      }
-    }
-    if (lastRecord != null) {
+      final InMemoryEngine engine, final ZeebeClient client, final String elementId) {
+    final List<Record<JobRecordValue>> records =
+        StreamFilter.jobRecords(engine.getRecordStream())
+            .withElementId(elementId)
+            .withIntent(JobIntent.CREATED)
+            .stream()
+            .collect(Collectors.toList());
+
+    if (!records.isEmpty()) {
+      final Record<JobRecordValue> lastRecord;
+      lastRecord = records.get(records.size() - 1);
```

Comment:
:x: Not clear to me why it would pick the last record. I would expect either the first (FIFO) or it completes all of them

Reply:
I just copied what we already had tbh. I think it is the last in case there is a loop. Than you wouldn't want to get the job you already completed earlier.

The real problem is that it uses this "hacky" and completes jobs without activating them first. I'll keep it like this for now, we can do it in a proper way when we implement the `TestClient`